lib/registry.js

```diff
@@ -124,21 +131,24 @@ class Registry {
 
 	async getMetricsAsJSON() {
 		const metrics = [];
-		const defaultLabelNames = Object.keys(this._defaultLabels);
+		let defaultLabelNames = Object.keys(this._defaultLabels);
+		if (defaultLabelNames.length === 0) {
+			defaultLabelNames = undefined;
+		}
 
 		const promises = [];
 
 		for (const metric of this.getMetricsAsArray()) {
 			promises.push(metric.get());
 		}
 
-		const resolves = await Promise.all(promises);
+		for (const promise of promises) {
+			const item = await promise;
```

Comment:
hmm, can't this result in unhandled rejections if one promise rejects while we're `await`ing another?

maybe we should `.then` chain it and keep the `Promise.all` or extract `metrics.get()` into a separate async function that also does the label thing?

Reply:
We will still blow up at the end, and we are forcing the event queue to lock up entirely once the promise.all() resolves. You’ve got a few people in the issues database claiming this library is unusable because of the delays and I tend to agree. 

Degraded stats are degrades stats. You have the same problem on the upstream aggregator. Dropouts make all of the data weird and unrepresentative. The thing I like least about OpenTelemetry is it’s all-or-nothing behavior. It’s not real when you have five hundred processes sending data, and two tiers of aggregation coalescing it. Which in Node might only be a couple dozen machines due to one process per core, one aggregator sidecar/pod per machine.

Reply:
Fixed.